### PR TITLE
At 201b

### PR DESCRIPTION
**Pull request by lmavridis**

### CODE REVIEW

File: R/postRecommendations.R [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+#' The function that return the n nearest types to the given ones
+#'
+#' @param type - the type (subcategory/category/type) to be checked
+#' @param data - the matrix of connections between the types
+#' @param n.of.types - the number of n nearest types
+#' @param exlude.same - remove the same type from the results
+returnTopTypes <- function(type, data, n.of.types = 3,
+                           exclude.same = TRUE) {
+  res <-  data[Type %in% type]
```
Comment (byapparov) [inline R/postRecommendations.R]: lower case

File: R/postRecommendations.R [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+  return (rec)
+}
+
+#' The function that return the n nearest types to the given ones
+#'
+#' @param type - the type (subcategory/category/type) to be checked
+#' @param data - the matrix of connections between the types
+#' @param n.of.types - the number of n nearest types
+#' @param exlude.same - remove the same type from the results
+returnTopTypes <- function(type, data, n.of.types = 3,
```
Comment (byapparov) [inline R/postRecommendations.R]: get..

File: tests/testthat/test.complimentarity.R [hunk 3]
```diff
@@ -0,0 +1,77 @@
+context("Post order functions.")
+
+test_that("Top types are correctly retrieved", {
+  
+  type <- "A"
+  data <- data.table(Type = c("A","B","C","D"),
```
Comment (byapparov) [inline tests/testthat/test.complimentarity.R]: what is this?

File: R/postRecommendations.R (hunk 1 above)
Comment (byapparov) [inline R/postRecommendations.R]: `data` is not specific enough

File: tests/testthat/test.complimentarity.R (hunk 3 above)
Comment (byapparov) [inline tests/testthat/test.complimentarity.R]: `data.table` does not make sense

File: R/postRecommendations.R [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+  
+  positions.to.retain <- (row.names(sim.matrix) %in% selected.skus$sku)
+  rec <- getSimilarProducts(sim.matrix[, positions.to.retain], skus, values, exclude.same = TRUE)
+
+  return (rec)
+}
+
+#' The function that return the n nearest types to the given ones
+#'
+#' @param type - the type (subcategory/category/type) to be checked
```
Comment (byapparov) [inline R/postRecommendations.R]: vector of types

File: tests/testthat/test.complimentarity.R (hunk 3 above)
Comment (byapparov) [inline tests/testthat/test.complimentarity.R]: Move function that creates `affiliate types data.table` to the package

File: R/postRecommendations.R (hunk 4 above)
Comment (lmavridis) [inline R/postRecommendations.R]: done

File: R/postRecommendations.R (hunk 2 above)
Comment (lmavridis) [inline R/postRecommendations.R]: Renamed

File: R/postRecommendations.R (hunk 1 above)
Comment (lmavridis) [inline R/postRecommendations.R]: done & done

File: tests/testthat/test.complimentarity.R (hunk 3 above)
Comment (lmavridis) [inline tests/testthat/test.complimentarity.R]: done

**Comment by lmavridis** [comment]
@byapparov Updated the pull request

File: R/postRecommendations.R [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+  return (res.type[1:n.of.types])
+}
+
+#' Given a list of transactions the post order affinity table is calculated.
+#' The expected format of transactions is a a data.table with
+#' UID - unique customer identifier
+#' reference - Grouping of orders
+#' type - type per item in an order
+#'
+#' @param transactions - a data.table with transaction data
```
Comment (byapparov) [inline R/postRecommendations.R]: @return?

File: R/lib.R [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+
+
+#' The function that drills in each order and associates the types
+#' based on the order history
+#' 
+#'
+#' @param list.of.orders - a single list of orders
+#' @param order.mapping - a single list of items per order
+connectOrderMapping <- function(list.of.orders, order.mapping){
+  sku1 <- NULL
```
Comment (lmavridis) [inline R/lib.R]: Done

**Pull request merged**